File: Challenge1/line.cpp
// Lines drawing algorithm
/*
Autor: 
Luis Alberto Alcántara Cabrales A01634185

Descripción:
Dado dos puntos en el formato (x,y) se dibuja 
una línea entre ellos utilizando el algoritmo 
de Bresenham.

Fecha: 1/11/2022
*/

#include <iostream>
#include <string>
using namespace std;

// Funcion que imprime en consola los puntos de la linea
void lineDrawing(int x1, int y1, int x2, int y2){
    int dx = x2 - x1;
    int dy = y2 - y1;

}

int main (){
    cout << "Lines drawing algorithm" << endl;

    // Variables
    int x1 = 0;
    int y1 = 0;
    int x2 = 0;
    int y2 = 0;

    // Entrada de datos
    string punto = " ";
    for (int i = 0; i < 2; i++){
        while(getline(cin, punto)){
            if (punto.length() > 0){
                break;
            }
        }
        if (i == 0){
            x1 = stoi(punto.substr(1, punto.find(",")));
            y1 = stoi(punto.substr(punto.find(",") + 1, punto.find(")")));
        } else {
            x2 = stoi(punto.substr(1, punto.find(",")));
            y2 = stoi(punto.substr(punto.find(",") + 1, punto.find(")")));
        }
    }

    // Llamada a la funcion
    lineDrawing(x1, y1, x2, y2);

    return 0;
}

File: Challenge2/line.js
// Drawing a line
/*
Autor:
Luis Alberto Alcántara Cabrales A01634185

Descripción:
Dibuja una linea entre dos puntos dados

Entradas:
x1, y1, x2, y2

Salidas:
Dibuja una linea entre los puntos dados

Fecha: 02/11/2022
*/

function setup() {
    createCanvas(400, 400);
}
  
function draw() {
    background(220);
    drawLine(1,2,100,20);
    drawLine(10,200,50,10);
    drawLine(300,300,10,10);
    drawLine(200,300,250,50);
    drawLine(350,350,100,300);
}
  
function drawLine(x1, y1, x2, y2) {
    var dx = x2 - x1;
    var dy = y2 - y1;
  
    var m = dy / dx;
    var b = y1 - m * x1;
  
    var x = x1;
    var y = y1;
  
    if (x2 > x1) {
        while (x <= x2) {
            point(x, y);
            x++;
            y = m * x + b;
        }
    } else {
        while (x >= x2) {
            point(x, y);
            x--;
            y = m * x + b;
        }
    }
}